### PR TITLE
Remove api client lib dependency on tlsconfig and sockets packages.

### PR DESCRIPTION
**Pull request by calavera**
- Let consumers to configure the http transport initially and apply or
  defaults on top.
- Add function to initialize a new client based on environment
  variables, useful for integrators.

/cc @dnephin

Signed-off-by: David Calavera david.calavera@gmail.com

### CODE REVIEW

File: api/client/cli.go [hunk 1]
```diff
@@ -108,7 +110,12 @@ func NewDockerCli(in io.ReadCloser, out, err io.Writer, clientFlags *cli.ClientF
 			verStr = tmpStr
 		}
 
-		client, err := lib.NewClient(host, verStr, clientFlags.Common.TLSOptions, customHeaders)
+		clientTransport, err := newClientTransport(clientFlags.Common.TLSOptinos)
```
Comment (MHBauer) [inline api/client/cli.go]: typo `TLSOptinos` should be `TLSOptions`. (At least I hope it should be)

**Comment by cpuguy83** [comment]
api/client/lib/client.go:53: unknown http.Transport field 'TLSConfig' in struct literal

**Comment by MHBauer** [comment]
Probably a good concept to be lower level, and less concerned about transport details.

That said, it is a `docker/pkg`, so it's not a foreign library we need to worry about depending on.

**Comment by calavera** [comment]
@MHBauer we're trying to not depend on `github.com/docker/docker` at all when we extract the library. The less dependencies the better, and people should not need to be familiar with hidden libraries that we use as helpers to use this new client.

**Comment by MHBauer** [comment]
I concur, might as well not pull the entirety of docker in if we don't have to.

File: api/client/lib/client.go [hunk 2]
```diff
@@ -32,11 +33,35 @@ type Client struct {
 	customHTTPHeaders map[string]string
 }
 
+// NewEnvClient initializes a new API client based on environment variables.
+// Use DOCKER_HOST to set the url to the docker server.
+// Use DOCKER_API_VERSION to set the version of the API to reach, leave empty for latest.
+// Use DOCKER_CERT_PATH to load the tls certificates from.
+func NewEnvClient() (*Client, error) {
```
Comment (cpuguy83) [inline api/client/lib/client.go]: This is being added but not used anywhere?
Also missing `DOCKER_TLS_VERIFY`

Comment (calavera) [inline api/client/lib/client.go]: that's correct, it won't be used yet, but other projects will use it for easy bootstrap. Good call about the verification, I'm going to change it.

Comment (calavera) [inline api/client/lib/client.go]: I've fixed the verification check already.

**Comment by tiborvass** [comment]
LGTM

**Comment by cpuguy83** [comment]
LGTM

**Pull request merged**